src/server/db_slice.cc

```diff
@@ -1338,4 +1361,57 @@ void DbSlice::ResetUpdateEvents() {
   events_.update = 0;
 }
 
+// todo: perhaps we need to limit the total number of entry in the tracking
+// table so that we have a way to limit the amount of memory used for
+// client tracking
+void DbSlice::TrackKeys(ConnectionContext* cntx, int32_t tid,
+                        const std::vector<std::string_view>& keys) {
+  DVLOG(2) << "Start tracking the following keys for thread ID: " << tid
+           << ", client ID: " << cntx->conn()->GetClientId();
+  for (auto key : keys) {
+    // std::string_view k = key;
+    std::string k{key.begin(), key.end()};
+    DVLOG(2) << "  " << k;
+    if (client_tracking_map_.find(k) == client_tracking_map_.end()) {
+      std::pair<ConnectionContext*, int32_t> p{cntx, tid};
+      absl::flat_hash_set<std::pair<ConnectionContext*, int32_t>, Hash> tracker_set{p};
+      client_tracking_map_.insert({k, tracker_set});
+    } else {
+      std::pair<ConnectionContext*, int32_t> p{cntx, tid};
+      client_tracking_map_[k].insert(p);
```

Comment:
1. Doesn't it work for the general case with [k]? I think that absl::flat_hash_set is default constructible
2. You don't need k to be a string, you can use [string_view]